### DISCUSSION TITLE
[Question] Referencing .NET Framework WinForms from .NET Core WinForms

### DISCUSSION THREAD

**Discussion by janseris** [dotnet/winforms#10455] [Q&A]
### .NET version

8.0.1

### Did it work in .NET Framework?

Not tested/verified

### Issue description

How is WinForms NET Core -> NET Framework intended to work?
I noticed that when I create old csproj style WinForms, Visual Studio does not report any warnings
When I switch to new csproj and change target framework moniker to .NET Framework 4.8, I start receiving warning which afaik cannot be disabled.
Using new csproj structure allows me to use Sync Namespaces feature in Visual Studio which is very important in large projects.
```

Severity	Code	Description	Project	File	Line	Suppression State
Warning	NU1702	ProjectReference 'C:\Users\Jan\Desktop\prace\WinFormsCsProjNewOldTest\WinFormsNetFrameworkClassLibraryNewProjectStructure\WinFormsNetFrameworkClassLibraryNewProjectStructure.csproj' was resolved using '.NETFramework,Version=v4.8' instead of the project target framework '.NETCoreApp,Version=v8.0'. This project may not be fully compatible with your project.	WinFormsNetCore	C:\Program Files\Microsoft Visual Studio\2022\Community\MSBuild\Current\Bin\amd64\Microsoft.Common.CurrentVersion.targets	1860	
```

Why is that? Is it supposed to be there?

![obrazek](https://github.com/dotnet/winforms/assets/64279914/2a857493-dd21-4784-94ee-d4c55e97a948)

### Steps to reproduce

https://github.com/janseris/WinFormsCsProjNewOldTest

**Comment by elachlan** [comment]
Hi @janseris,

There are a couple of options listed below
1. Multi-target. Specify multiple targets under the `TargetFrameworks` property. See: https://learn.microsoft.com/en-us/nuget/create-packages/multiple-target-frameworks-project-file
```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <TargetFrameworks>net6.0;net48</TargetFrameworks>
    <Description>Sample project that targets multiple TFMs</Description>
  </PropertyGroup>

</Project>
```
2. Use the `netstandard2.0` `TargetFramework` for your class library
3. Migrate everything to .NET core
4. Stick with .net4.8

Hope that helps.

**Comment by janseris** [comment]
> Hi @janseris,
> 
> There are a couple of options listed below
> 
>     1. Multi-target. Specify multiple targets under the `TargetFrameworks` property. See: https://learn.microsoft.com/en-us/nuget/create-packages/multiple-target-frameworks-project-file
> 
> 
> ```
> <Project Sdk="Microsoft.NET.Sdk">
> 
>   <PropertyGroup>
>     <TargetFrameworks>net6.0;net48</TargetFrameworks>
>     <Description>Sample project that targets multiple TFMs</Description>
>   </PropertyGroup>
> 
> </Project>
> ```
> 
>     2. Use the `netstandard2.0` `TargetFramework` for your class library
> 
>     3. Migrate everything to .NET core
> 
>     4. Stick with .net4.8
> 
> 
> Hope that helps.

But why is the warning dependent on just having different style of csproj? 
I am working without any issues referencing .NET Framework WinForms Control library from a .NET Core project in production without multitargeting.

**Comment by elachlan** [comment]
I have no idea. But if you want to have both .net framework and .net core projects, you either multi-target or use .netstandard 2.0.

You might have better luck getting a more in-depth answer on https://stackoverflow.com/

**Comment by merriemcgaw** [comment]
I am going to convert this issue to a discussion for further conversation, but I agree with @elachlan that your best bet might be another forum.
